src/telemetry/metrics/prom.rs

```diff
@@ -0,0 +1,126 @@
+use std::fmt;
+use std::marker::PhantomData;
+
+/// Writes a block of metrics in prometheus-formatted output.
+pub trait FmtPrometheus {
```

Comment:
I feel like the `FmtPrometheus` and `FmtMetric` trait names aren't super clear --- the comments help a bit, but just seeing the trait names, it's not immediately clear which does what. It might be good if that could be made clearer...